src/protocols/interaction_model/StatusCode.h

```diff
@@ -38,7 +38,7 @@ namespace Protocols {
 namespace InteractionModel {
 
 // This table comes from the IM's "Status Code Table" section from the Interaction Model spec.
-enum class Status : uint8_t
+enum Status : uint8_t
```

Comment:
Again: there should be a comment here (in the code, not in the PR review thread) with the issue link.
```suggestion
// TODO: This needs to go back to being an enum class.  https://github.com/project-chip/connectedhomeip/issues/32025
enum Status : uint8_t
```